src/Pipelines/Stages/ReIndexStage.php

```diff
@@ -54,6 +55,18 @@ public function __invoke($payload)
                     ],
                 ],
             ]);
+
+            for (; ;) {
```

Comment:
I would change this to a do/while loop where the condition is `(bool)$response['completed']`